### PR TITLE
Feature/84 improve get form validation result.field errors

### PR DESCRIPTION
**Pull request by nasdan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,10 +43,10 @@ customerFormValidation
...
     console.log(validationResult.formGlobalErrors); // []
     console.log(validationResult.fieldErrors);
-    /*[
-        { succeeded: true, type: "REQUIRED", key: "firstName", errorMessage: "" },
-        { succeeded: true, type: "REQUIRED", key: "lastName", errorMessage: "" }
-      ]*/
+    /*{
+        firstName: { succeeded: true, type: "REQUIRED", key: "firstName", errorMessage: "" },
+        lastName: { succeeded: true, type: "REQUIRED", key: "lastName", errorMessage: "" }
+      }*/
```
Comment (brauliodiez) [inline README.md:49]: I don't remember what was the intention when we create the readme about this commented code, dou you know why?

Comment (brauliodiez) [reply README.md:49]: I see it's dumping the result, it's from a previous version (even previous that current)

File: lib/src/entities.ts [hunk 2]
```diff
@@ -24,12 +24,12 @@ export class FieldValidationResult {
...
 export class FormValidationResult {
   succeeded: boolean;
-  fieldErrors: FieldValidationResult[];
+  fieldErrors: { [key: string]: FieldValidationResult };
   formGlobalErrors: FieldValidationResult[];
 
   constructor() {
     this.succeeded = false;
-    this.fieldErrors = [];
+    this.fieldErrors = {};
```
Comment (brauliodiez) [inline lib/src/entities.ts:32]: MMM... I think this is a breaking change, should we go for a major release?

Comment (nasdan) [reply lib/src/entities.ts:32]: Yes, I think it should be major

**Pull request merged**